File: java/com/countrygamer/countrygamercore/client/ClientProxy.java
package com.countrygamer.countrygamercore.client;

import net.minecraft.entity.player.EntityPlayer;

import com.countrygamer.countrygamercore.common.CommonProxy;
import com.countrygamer.countrygamercore.common.Core;
import com.countrygamer.countrygamercore.common.network.MessageSyncExtendedProperties;
import com.countrygamer.countrygamercore.common.network.PacketHandler;

import cpw.mods.fml.client.registry.RenderingRegistry;
import cpw.mods.fml.common.FMLCommonHandler;

public class ClientProxy extends CommonProxy {
	
	@Override
	public void registerRender() {
		RenderingRegistry.registerBlockHandler(new BlockCamouflageRender());
		
	}
	
	@Override
	public void syncPacket(MessageSyncExtendedProperties message, EntityPlayer player) {
		if (FMLCommonHandler.instance().getEffectiveSide().isClient()) {
			PacketHandler.sendToServer(Core.pluginID, message);
		}
		else {
			super.syncPacket(message, player);
		}
	}
	
	@Override
	public int addArmor(String armor) {
		return RenderingRegistry.addNewArmourRendererPrefix(armor);
	}
	
}

File: src/main/scala/com/countrygamer/cgo/client/ClientProxy.scala
package com.countrygamer.cgo.client

import com.countrygamer.cgo.client.render.BlockCamouflageRender
import com.countrygamer.cgo.common.network.{PacketHandler, PacketSyncExtendedProperties}
import com.countrygamer.cgo.common.{CommonProxy, Origin}
import cpw.mods.fml.client.registry.RenderingRegistry
import cpw.mods.fml.common.FMLCommonHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.world.World

/**
 *
 *
 * @author CountryGamer
 */
class ClientProxy() extends CommonProxy {

	override def registerRender(): Unit = {
		RenderingRegistry.registerBlockHandler(BlockCamouflageRender)

	}

	override def getClientGuiElement(p1: Int, p2: EntityPlayer, p3: World, p4: Int, p5: Int,
			p6: Int): AnyRef = {
		null
	}

	override def syncPacket(message: PacketSyncExtendedProperties, player: EntityPlayer) {
		if (FMLCommonHandler.instance.getEffectiveSide.isClient) {
			PacketHandler.sendToServer(Origin.pluginID, message)
		}
		else {
			super.syncPacket(message, player)
		}
	}

	override def addArmor(armor: String): Int = {
		RenderingRegistry.addNewArmourRendererPrefix(armor)
	}

}
